arangod/Cluster/ClusterInfo.cpp

```diff
@@ -6600,53 +6670,87 @@ ClusterInfo::getCurrent(uint64_t& index,
 
 containers::FlatHashSet<ServerID> ClusterInfo::getFailedServers() const {
   std::lock_guard lock{_failedServersMutex};
-  return _failedServers;
+  containers::FlatHashSet<ServerID> result;
+  result.reserve(_failedServers.size());
+  for (auto const& server : _failedServers) {
+    result.emplace(server);
+  }
+  return result;
 }
 
 void ClusterInfo::setFailedServers(
     containers::FlatHashSet<ServerID> failedServers) {
   std::lock_guard lock{_failedServersMutex};
-  _failedServers = std::move(failedServers);
+  _failedServers.clear();
+  for (auto const& server : failedServers) {
+    _failedServers.emplace(server);
+  }
```

Comment:
nit: this method gets its argument by value, so we could now move the values into the container.
```suggestion
  for (auto& server : failedServers) {
    _failedServers.emplace(std::move(server));
  }
```

Reply:
no we can't, because the string inside the container uses a different allocator.